<a href="https://colab.research.google.com/github/Prathmon/Class123/blob/main/Class122.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
import numpy as np
import pandas as pd
import seaborn as sns
import cv2
import matplotlib.pyplot as plt
from sklearn.datasets import fetch_openml
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score
from PIL import Image
import PIL.ImageOps 
import os, ssl, time

1. Import cv2 - This is the library with which we will use the computer's camera
2. Import numpy as np - With this we can perform complex mathematical list operations
3. Import pandas as pd- With this we can read the data in the form of dataframe
4. Import seaborn as sns - We use this module to prettify the charts that we draw with matplotlib
5. Import matplotlib.pyplot as plt - This module is used to draw charts.
6. from sklearn.datasets import fetch_openml - This function allows us to retrieve a dataset by name from openml. Openml is a public repository from where we can import 
7. from sklearn.model_selection import train_test_split - This is used to split the data into training and testing.
8. from sklearn.linear_model import LogisticRegression - It is used to create a logistic regression classifier.
9. from sklearn.metrics import accuracy_score - It is used to test the accuracy score of a model

In [11]:
if (not os.environ.get('PYTHONHTTPSVERIFY', '') and
    getattr(ssl, '_create_unverified_context', None)):
    ssl._create_default_https_context = ssl._create_unverified_context

Setting an https context to fetch data from openml

Fetching the data
X, y

In [12]:
X, y = fetch_openml('mnist_784', version = 1, return_X_y = True)
print(pd.Series(y).value_counts())
classes = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']
nclasses = len(classes)

1    7877
7    7293
3    7141
2    6990
9    6958
0    6903
6    6876
8    6825
4    6824
5    6313
Name: class, dtype: int64


As we know that we can represent an image in binary, here in the first line of code we are fetching the dataset of images of handwritten digits from the openml datasets. Here X is data of images represented in binary, y is label of that image. Next, we are counting the samples for each of the labels. As in the output we can see that label 1 has 7877, label 2 has 6990 and so on. Finally, we are creating a list that contains all the labels of the images.

In [ ]:
samples_per_class = 5
figure = plt.figure(figsize = (nclasses*2, (1+samples_per_class*2)))
print(figure)

Figure(1440x792)


<Figure size 1440x792 with 0 Axes>

Splitting the data and scaling it

In [13]:
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state = 9, train_size = 7500, test_size = 2500)
X_train_scaled = X_train/255.0
X_test_scaled = X_test/255.0

Fitting the training data into model

In [19]:
clf = LogisticRegression(solver='saga', multi_class='multinomial').fit(X_train_scaled, y_train)
y_pred = clf.predict(X_test_scaled)
accuracy = accuracy_score(y_test, y_pred)
print('Accuracy is:', accuracy)

Accuracy is: 0.9


/usr/local/lib/python3.7/dist-packages/sklearn/linear_model/_sag.py:354: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  ConvergenceWarning,


Scaling the features